### PR TITLE
#32: Fix nodejs version

### LINKED ISSUE #32: Update to NodeJs 14
`terraform apply` displays the error "_The runtime parameter of nodejs12.x is no longer supported._"

The current ticket therefore requests to update to NodeJs 14.
Please note: NodeJS 18 already requires AWS SDK 2.0 containing breaking changes.

Currently `main.tf` references cloud formation template `cloudformation_template_v0.0.3.yml` with in turn uses NodeJs 12.
So the content of this resource needs to be updated.

### AC
1. old versions are kept in Git repo terraform-aws-exasol and in S3 bucket for backwards compatibility 
2. a new version 0.0.4 is added 
3. new version 0.0.4 uses NodeJS 14

### PR DESCRIPTION
**Pull request by kaklakariada**
Fixes #32

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -9,52 +9,23 @@ jobs:
...
       - name: Install terraform
-        run: |
-          wget -O terraform.zip "https://releases.hashicorp.com/terraform/${{ matrix.terraform_version }}/terraform_${{ matrix.terraform_version }}_linux_amd64.zip"
-          echo "${{ matrix.sha256sum }}  terraform.zip" | sha256sum -c
-          unzip terraform.zip
-          rm -f terraform.zip
-          chmod +x terraform
-          mv terraform ${HOME}/bin/
-          terraform -v
+        uses: hashicorp/setup-terraform@v2
```
Comment (redcatbear) [inline .github/workflows/ci-build.yml:21]: Better. :+1:

File: main.tf [hunk 2]
```diff
@@ -15,8 +15,7 @@ resource "aws_cloudformation_stack" "exasol_cluster" {
   capabilities = [
   "CAPABILITY_IAM"]
   on_failure   = "DELETE"
-  template_url = "https://exasol-cf-templates.s3.eu-central-1.amazonaws.com/cloudformation_template_v0.0.3.yml"
-
+  template_url = "https://exasol-cf-templates.s3.eu-central-1.amazonaws.com/cloudformation_template_v1.0.3.yml"
```
Comment (redcatbear) [inline main.tf:18]: Unlike the old link, I cannot download from this one. Extra protection, future link or broken?

**Review by redcatbear — DISMISSED** [review]

File: main.tf (hunk 2 above)
Comment (kaklakariada) [reply main.tf:18]: Good catch!
This is a future link. Need to get permissions for deployment first.

Comment (redcatbear) [reply main.tf:18]: :heavy_check_mark:

File: doc/developers_guide/developers_guide.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+### Running System Tests
+
+* Install Terraform and Go
+* Get an Exasol license and store it as `test/simple_exasol_setup/exasolution.lic`
+* Create the file `test/simple_exasol_setup/terraform.tfvars` and fill in:
+  ```
+  owner = "<YOUR_EMAIL>"
+  aws_profile = "<AWS PROFILE TO USE>"
+  ```
+* Run `cd test && go test --timeout 2h`
```
Comment (redcatbear) [inline doc/developers_guide/developers_guide.md:26]: Move to separate command box. Easier to copy.

File: doc/developers_guide/developers_guide.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+## Publishing to Terraform Registry
+
+To publish the module to the Terraform registry just create a release in the GitHub repository.
```
Comment (redcatbear) [inline doc/developers_guide/developers_guide.md:30]:
```suggestion
To publish the module to the Terraform registry create a release in the GitHub repository.
```

**Review by redcatbear — CHANGES_REQUESTED** [review]

**Review by redcatbear — APPROVED** [review]

**Review by redcatbear — APPROVED** [review]

**Pull request merged**